pkg/reconciler/v1alpha1/pipelinerun/resources/pipelinerunresolution_test.go

```diff
@@ -853,3 +847,18 @@ func TestValidateFrom_Invalid(t *testing.T) {
 		})
 	}
 }
+
+type testGetter struct {
+	err error
+	r   *v1alpha1.PipelineResource
+}
+
+func (t testGetter) Get(n string) (*v1alpha1.PipelineResource, error) {
```

Comment:
why did this change? (seems like it makes the interface more complex, but im not seeing why we need it?)

Reply:
Previously interface assumed that pipeline resource is always fetcher. Now there is fetch if reference is present or consume the spec directly.

Reply:
hm I'm still a bit confused b/c when I look at how the getter is used, ultimately it looks like it ends up being:

```
	if r.ResourceRef.Name != "" {
		return getter.Get(r.ResourceRef.Name)
	}
```

I'm not seeing any other uses of `gettter`, i.e. it seems like we could continue to pass the `Get` function in directly? (not sure why we need the whole lister now)

(sorry im probably still missing something!)

Reply:
My bad. I think I understand what you are saying. I wanted to either use `ResourceGetter`(with Get func) or Get func interface but not both. Initially I took the approach to have all resource resolution functions use `ResourceGetter` interface but I see your point that it is making code and tests more complicated. I changed the approach to use `Get` func directly. Let me know what you think @bobcatfish